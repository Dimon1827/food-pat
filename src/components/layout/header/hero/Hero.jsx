import { useNavigate } from 'react-router-dom';
import Button from '../../../ui/button/Button';
import './Hero.scss';

function Hero() {
  const navigate = useNavigate();
  return (
    <section className="hero">
      <div className="container">
        <div className="hero__wrapper">
          <h1 className="hero__title title">Магазин здорового питания</h1>
          <p className="hero__text">Поможем подобрать оптимальную программу</p>
          <picture>
            <source
              type="image/webp"
              media="(min-width:1280px)"
              srcSet="
                    ./hero-desktop@1x.webp,
                    ./hero-desktop@2x.webp 2x
                  "
              width="552"
              height="499"
            />
            <source
              type="image/webp"
              media="(min-width:768px)"
              srcSet="
                    ./hero-tablet@1x.webp,
                    ./hero-tablet@2x.webp 2x
                  "
              width="300"
              height="300"
            />
            <source
              type="image/webp"
              srcSet="
                    ./hero-mobile@1x.webp,
                    ./hero-mobile@2x.webp 2x
                  "
              width="280"
              height="270"
            />
            <source
              type="image/png"
              media="(min-width:1280px)"
              srcSet="
                    ./hero-desktop@1x.png,
                    ./hero-desktop@2x.png 2x
                  "
              width="552"
              height="499"
            />
            <source
              type="image/png"
              media="(min-width:768px)"
              srcSet="
                    ./hero-tablet@1x.png,
                    ./hero-tablet@2x.png 2x
                  "
              width="300"
              height="300"
            />
            <img
              src="./hero-mobile@1x.png"
              srcSet="./hero-mobile@2x.png 2x"
              width="280"
              height="270"
              alt="Супергерой"
              className="hero__image"
            />
          </picture>
          <Button onClick={() => navigate('/form')} className="hero__button">
            подобрать
          </Button>
        </div>
      </div>
    </section>
  );
}

export default Hero;
