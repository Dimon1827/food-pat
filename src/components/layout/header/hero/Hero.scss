@import 'src/assets/styles/variables';

.hero {
  position: relative;
  background: linear-gradient(180deg, $bg-basic-orange-3 100%, $white-color 40%);

  padding-bottom: 20px;

  @media (min-width: $tablet-width) {
    background: none;
    padding-bottom: 0;
  }

  .hero__title {
  color: $white-color;
  margin-bottom: 25px;

  @media (min-width: $tablet-width) {
    width: 425px;
    order: -3;
    margin-bottom: 40px;
  }

  @media (min-width: $desktop-width) {
    margin-left: 79px;
    color: $black-color;
  }
}
}

.hero__wrapper {
  padding-top: 27px;
  text-align: center;
  margin-bottom: 20px;
  display: flex;
  flex-direction: column;

  @media (min-width: $tablet-width) {
    position: relative;
    text-align: left;
    margin-bottom: 0;
    padding: 40px;
    background: $bg-basic-orange-3;
  }

  @media (min-width: $desktop-width) {
    background: transparent;
  }
}



.hero__text {
  font-family: $oswald-font;
  margin: 0;
  line-height: 14px;
  color: $white-color;
  text-transform: uppercase;
  margin-bottom: 31px;

  @media (min-width: $tablet-width) {
    font-size: 20px;
    margin-bottom: 50px;
    order: -2;
    max-width: 283px;
    line-height: 155%;
  }

  @media (min-width: $desktop-width) {
    color: $black-color;
    margin-left: 79px;
    margin-bottom: 53px;
  }
}

.hero__button {
  width: 100%;
  max-width: 385px;
  margin-left: auto;
  margin-right: auto;

  @media (min-width: $tablet-width) {
    position: relative;
    z-index: 1;
    font-size: 20px;
    line-height: 20px;
    order: -1;
    max-width: 244px;
    margin-left: 0;
    padding: 16px 8px;
  }

  @media (min-width: $desktop-width) {
    margin-left: 79px;
  }
}

.hero__image {
  margin: 0 auto;

  @media (min-width: $tablet-width) {
    position: absolute;
    left: 49%;
    top: 186px;
    margin-top: -82px;
  }

  @media (min-width: $desktop-width) {
    left: 584px;
    top: -10px;
  }
}
