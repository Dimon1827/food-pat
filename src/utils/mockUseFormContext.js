import { useFormContext } from "react-hook-form";

export function mockUseFormContext(errors) {
  useFormContext.mockReturnValue({
    register: jest.fn(),
    formState: {
      errors: errors,
    },
  });
}


