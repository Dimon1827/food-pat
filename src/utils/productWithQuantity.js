const productWithQuantity = (food) => {
  return {
    ...food,
    isChecked: false,
    quantity: 1,
  };
};

export default productWithQuantity;
