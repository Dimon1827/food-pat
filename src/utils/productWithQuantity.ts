import { IProducts } from "redux/slice/productsSlice.types";

interface IproductWithQuantity extends Omit<IProducts, 'isChecked' | 'quantity'> {};

const productWithQuantity = (food: IproductWithQuantity) => {
  return {
    ...food,
    isChecked: false,
    quantity: 1,
  };
};

export default productWithQuantity;
