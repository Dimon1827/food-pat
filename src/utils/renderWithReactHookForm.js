import { render } from '@testing-library/react';
import { FormProvider, useForm } from 'react-hook-form';

export function renderWithReactHookForm(ui, { defaultValues = {} } = {}) {
  let reactHookFormMethods = {};

  const Wrapper = ({ children }) => {
    const methods = useForm({ defaultValues });
    return <FormProvider {...methods}>{children}</FormProvider>;
  };

  return {
    ...render(ui, { wrapper: Wrapper }),
  };
}