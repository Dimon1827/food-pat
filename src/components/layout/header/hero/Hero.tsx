import { useNavigate } from 'react-router-dom';
import Button from '@/components/ui/button/Button';
import './Hero.scss';

function Hero() {
  const navigate = useNavigate();
  return (
    <section className="hero" data-testid = "Hero">
      <div className="container">
        <div className="hero__wrapper">
          <h1 className="hero__title title">Магазин здорового питания</h1>
          <p className="hero__text">Поможем подобрать оптимальную программу</p>
          <picture>
            <source
              type="image/webp"
              media="(min-width:1280px)"
              srcSet="
                    ./assets/images/hero-desktop@1x.webp,
                    ./assets/images/hero-desktop@2x.webp 2x
                  "
              width="552"
              height="499"
            />
            <source
              type="image/webp"
              media="(min-width:768px)"
              srcSet="
                    ./assets/images/hero-tablet@1x.webp,
                    ./assets/images/hero-tablet@2x.webp 2x
                  "
              width="300"
              height="300"
            />
            <source
              type="image/webp"
              srcSet="
                    ./assets/images/hero-mobile@1x.webp,
                    ./assets/images/hero-mobile@2x.webp 2x
                  "
              width="280"
              height="270"
            />
            <source
              type="image/png"
              media="(min-width:1280px)"
              srcSet="
                    ./assets/images/hero-desktop@1x.png,
                    ./assets/images/hero-desktop@2x.png 2x
                  "
              width="552"
              height="499"
            />
            <source
              type="image/png"
              media="(min-width:768px)"
              srcSet="
                    ./assets/images/hero-tablet@1x.png,
                    ./assets/images/hero-tablet@2x.png 2x
                  "
              width="300"
              height="300"
            />
            <img
              src="./assets/images/hero-mobile@1x.png"
              srcSet="./assets/images/hero-mobile@2x.png 2x"
              width="280"
              height="270"
              alt="Супергерой"
              className="hero__image"
            />
          </picture>
          <Button onClick={() => navigate('/form')} className="hero__button">
            подобрать
          </Button>
        </div>
      </div>
    </section>
  );
}

export default Hero;
