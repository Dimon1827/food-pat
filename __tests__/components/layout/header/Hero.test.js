import { screen } from '@testing-library/react';
import { renderWithRouter } from '@/utils/renderWithRouter';
import Hero from '@/components/layout/header/hero/Hero';

describe('Hero', () => {
  it('Hero renders correctly', () => {
    renderWithRouter(<Hero />);
    expect(screen.getByTestId('Hero')).toMatchSnapshot();
  });
});
