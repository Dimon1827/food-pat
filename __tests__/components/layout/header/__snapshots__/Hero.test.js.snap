// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Hero renders correctly 1`] = `
<section
  class="hero"
  data-testid="Hero"
>
  <div
    class="container"
  >
    <div
      class="hero__wrapper"
    >
      <h1
        class="hero__title title"
      >
        Магазин здорового питания
      </h1>
      <p
        class="hero__text"
      >
        Поможем подобрать оптимальную программу
      </p>
      <picture>
        <source
          height="499"
          media="(min-width:1280px)"
          srcset="
                    ./assets/images/hero-desktop@1x.webp,
                    ./assets/images/hero-desktop@2x.webp 2x
                  "
          type="image/webp"
          width="552"
        />
        <source
          height="300"
          media="(min-width:768px)"
          srcset="
                    ./assets/images/hero-tablet@1x.webp,
                    ./assets/images/hero-tablet@2x.webp 2x
                  "
          type="image/webp"
          width="300"
        />
        <source
          height="270"
          srcset="
                    ./assets/images/hero-mobile@1x.webp,
                    ./assets/images/hero-mobile@2x.webp 2x
                  "
          type="image/webp"
          width="280"
        />
        <source
          height="499"
          media="(min-width:1280px)"
          srcset="
                    ./assets/images/hero-desktop@1x.png,
                    ./assets/images/hero-desktop@2x.png 2x
                  "
          type="image/png"
          width="552"
        />
        <source
          height="300"
          media="(min-width:768px)"
          srcset="
                    ./assets/images/hero-tablet@1x.png,
                    ./assets/images/hero-tablet@2x.png 2x
                  "
          type="image/png"
          width="300"
        />
        <img
          alt="Супергерой"
          class="hero__image"
          height="270"
          src="./assets/images/hero-mobile@1x.png"
          srcset="./assets/images/hero-mobile@2x.png 2x"
          width="280"
        />
      </picture>
      <button
        class="button hero__button"
      >
        подобрать
      </button>
    </div>
  </div>
</section>
`;
